Invalid access: Can not convert empty value

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
daily run window function and the failed sql is  `select i, i, row_number()  over w from window_function_0 window w as (partition by i order by id, i) order by 1, 2, 3`, the full error info is `Error 1105: other error for mpp stream: From MPP<query:<query_ts:0, local_query_id:0, server_id:0, start_ts:438529181194125313>,task_id:1>: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = Invalid access: Can not convert empty value., e.what() = Invalid access`.
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiFlash version? (Required)

I encountered this panic too.
```
[2023/01/05 18:01:29.371 +08:00] [ERROR] [Exception.cpp:89] ["Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = Invalid access: Can not convert empty value., e.what() = Invalid access"] [source="DB::RawCppPtr DB::PreHandleSnapshot(DB::EngineStoreServerWrap *, DB::BaseBuffView, uint64_t, DB::SSTViewVec, uint64_t, uint64_t)"] [thread_id=23]
```

```
[source="DB::RawCppPtr DB::PreHandleSnapshot(DB::EngineStoreServerWrap *, DB::BaseBuffView, uint64_t, DB::SSTViewVec, uint64_t, uint64_t)"] [thread_id=50]
[2023/01/14 00:58:01.071 +08:00] [INFO] [TiDBSchemaSyncer.h:113] ["Start to sync schemas. current version is: 34 and try to sync schema version to: 39"] [source=SchemaSyncer] [thread_id=51]
[2023/01/14 00:58:01.071 +08:00] [DEBUG] [TiDBSchemaSyncer.h:171] ["Try load schema diffs."] [source=SchemaSyncer] [thread_id=51]
[2023/01/14 00:58:01.071 +08:00] [ERROR] [Exception.cpp:89] ["Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = Invalid access: Can not convert empty value., e.what() = Invalid access"] [source="DB::RawCppPtr DB::PreHandleSnapshot(DB::EngineStoreServerWrap *, DB::BaseBuffView, uint64_t, DB::SSTViewVec, uint64_t, uint64_t)"] [thread_id=51]
```

--- a/dbms/src/TiDB/Schema/SchemaGetter.cpp
+++ b/dbms/src/TiDB/Schema/SchemaGetter.cpp
@@ -156,32 +156,37 @@ void AffectedOption::deserialize(Poco::JSON::Object::Ptr json)
 void SchemaDiff::deserialize(const String & data)
 {
     Poco::JSON::Parser parser;
-    Poco::Dynamic::Var result = parser.parse(data);
-    if (result.isEmpty())
-    {
-        throw Exception("The schema diff deserialize failed " + data);
-    }
-    auto obj = result.extract<Poco::JSON::Object::Ptr>();
-    version = obj->getValue<Int64>("version");
-    type = static_cast<SchemaActionType>(obj->getValue<Int32>("type"));
-    schema_id = obj->getValue<Int64>("schema_id");
-    table_id = obj->getValue<Int64>("table_id");
-
-    old_table_id = obj->getValue<Int64>("old_table_id");
-    old_schema_id = obj->getValue<Int64>("old_schema_id");
-
-    regenerate_schema_map = obj->getValue<bool>("regenerate_schema_map");
-
-    affected_opts.clear();
-    auto affected_arr = obj->getArray("affected_options");
-    if (!affected_arr.isNull())
-    {
-        for (size_t i = 0; i < affected_arr->size(); i++)
+    try {
+        Poco::Dynamic::Var result = parser.parse(data);
+        if (result.isEmpty())
         {
-            auto affected_opt_json = affected_arr->getObject(i);
-            AffectedOption affected_option(affected_opt_json);
-            affected_opts.emplace_back(affected_option);
+            throw Exception("The schema diff deserialize failed " + data);
         }
+        auto obj = result.extract<Poco::JSON::Object::Ptr>();
+        version = obj->getValue<Int64>("version");
+        type = static_cast<SchemaActionType>(obj->getValue<Int32>("type"));
+        schema_id = obj->getValue<Int64>("schema_id");
+        table_id = obj->getValue<Int64>("table_id");
+
+        old_table_id = obj->getValue<Int64>("old_table_id");
+        old_schema_id = obj->getValue<Int64>("old_schema_id");
+
+        regenerate_schema_map = obj->getValue<bool>("regenerate_schema_map");
+
+        affected_opts.clear();
+        auto affected_arr = obj->getArray("affected_options");
+        if (!affected_arr.isNull())
+        {
+            for (size_t i = 0; i < affected_arr->size(); i++)
+            {
+                auto affected_opt_json = affected_arr->getObject(i);
+                AffectedOption affected_option(affected_opt_json);
+                affected_opts.emplace_back(affected_option);
+            }
+        }
+    } catch (...) {
+        LOG_INFO(&Poco::Logger::get("SchemaDiff"), "failed to deserialize {}", data);
+        throw;
     }
 }
 

--- a/dbms/src/TiDB/Schema/SchemaGetter.cpp
+++ b/dbms/src/TiDB/Schema/SchemaGetter.cpp
@@ -156,7 +156,8 @@ void AffectedOption::deserialize(Poco::JSON::Object::Ptr json)
 void SchemaDiff::deserialize(const String & data)
 {
     Poco::JSON::Parser parser;
-    try {
+    try
+    {
         Poco::Dynamic::Var result = parser.parse(data);
         if (result.isEmpty())
         {
@@ -184,7 +185,9 @@ void SchemaDiff::deserialize(const String & data)
                 affected_opts.emplace_back(affected_option);
             }
         }
-    } catch (...) {
+    }
+    catch (...)
+    {
         LOG_INFO(&Poco::Logger::get("SchemaDiff"), "failed to deserialize {}", data);
         throw;
     }

